hashicorp/consul-k8s
# Webhook refactor

Refactor the connect-inject webhook to use the `admission.Webhook` library for it's behavior. The behavior of the webhook should remain unchanged. The difference now is that the webhook does not manage its own certificates. Rather it relies on the webhook cert manager to provision certificates for it as a mounted secret.

This allows us to remove our dependency from DiskCerts and also remove the code involved with cert generation on the connect webhook. This is hopefully allow for the a degree of HA with the deployment as the webhook certificates will be consistent across the pods and webhook configuration.

I have also updated some of the methods within the connect package that previously has method signatures with (*Pod) -> (Pod) as they were not performing modifications on the Pod object.

How to test this PR: Code Review

There will be a companion Helm PR where these changes shall be acceptance tested and will try and ensure behavior is still the same.
